templates/demo/app/components/blockchain.js

```diff
@@ -0,0 +1,96 @@
+import EmbarkJS from 'Embark/EmbarkJS';
+import SimpleStorage from 'Embark/contracts/SimpleStorage';
+import React from 'react';
+import { Form, FormGroup, FormControl, HelpBlock, Button } from 'react-bootstrap';
+ 
+class Blockchain extends React.Component {
+
+    constructor(props) {
+      super(props);
+  
+      this.setValue = this.setValue.bind(this);
+      this.getValue = this.getValue.bind(this);
+      this.handleChange = this.handleChange.bind(this);
+  
+      this.state = {
+        valueSet: 10,
+        valueGet: "",
+        logs: []
+      }
+    }
+  
+    handleChange(e){
+      this.setState({valueSet: e.target.value});
+    }
+  
+    setValue(e){
+      e.preventDefault();
+  
+      var value = parseInt(this.state.valueSet, 10);
+  
+      // If web3.js 1.0 is being used
+      if (EmbarkJS.isNewWeb3()) {
+        SimpleStorage.methods.set(value).send({from: web3.eth.defaultAccount});
+        this._addToLog("SimpleStorage.methods.set(value).send({from: web3.eth.defaultAccount})");
+      } else {
+        SimpleStorage.set(value);
+        this._addToLog("#blockchain", "SimpleStorage.set(" + value + ")");
+      }
+    }
+  
+    getValue(e){
+      e.preventDefault();
+      
+      if (EmbarkJS.isNewWeb3()) {
+        SimpleStorage.methods.get().call()
+          .then(_value => this.setState({valueGet: _value}))
+        this._addToLog("SimpleStorage.methods.get(console.log)");
+      } else {
+        SimpleStorage.get()
+          .then(_value => this.setState({valueGet: _value}));
+        this._addToLog("SimpleStorage.get()");
+      }
+    }
+  
+    _addToLog(txt){
+      this.state.logs.push(txt);
+      this.setState({logs: this.state.logs});
+    }
+  
+    render(){
+      return (<React.Fragment>
+          <h3> 1. Set the value in the blockchain</h3>
+          <Form inline>
+            <FormGroup>
+              <FormControl
+                type="text"
+                defaultValue={this.state.valueSet}
+                onChange={this.handleChange}/>
+              {' '}
```

Comment:
What is this for? If it is just to add a spacer, maybe use a css margin?

Reply:
Fixed: Added a margin-right to input.form-control in dapp.css to avoid having to use spacers